sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/AbstractMobileCenterService.java

```diff
@@ -310,4 +312,44 @@ public void run() {
             disabledOrNotStartedRunnable.run();
         }
     }
+
+    /**
+     * Like {{@link #post(Runnable)}} but also post back in U.I. thread.
+     * Use this for example to manage life cycle callbacks to make sure SDK is started and that
+     * every operation runs in order.
+     *
+     * @param runnable command to run.
+     */
+    protected synchronized void postOnUiThread(final Runnable runnable) {
+
+        /*
+         * We don't try to optimize with if channel if not null as there could be race conditions:
+         * If onResume was queued, then onStarted called, onResume will be next in queue and thus
+         * onPause could be called between the queued onStarted and the queued onResume.
+         */
+        post(new Runnable() {
+
+            @Override
+            public void run() {
+
+                /* And make sure we run the original command on U.I. thread. */
+                HandlerUtils.runOnUiThread(new Runnable() {
+
+                    @Override
+                    public void run() {
+                        runIfEnabled(runnable);
+                    }
+                });
+            }
+        });
+    }
+
+    /**
+     * The method is top level just because code coverage when using synchronized.
```

Comment:
Fixing.
But we'll need to filter Javadoc to the classes that are not SDK internals.